pkgs/development/interpreters/octave/default.nix

```
@@ -90,121 +92,147 @@ let
+    inherit pkgs;
+    inherit (pkgs) lib stdenv fetchurl newScope;
+    octave = self;
   };
 
-  buildInputs = [
-    readline
-    ncurses
-    perl
-    flex
-    qhull
-    graphicsmagick
-    pcre
-    fltk
-    zlib
-    curl
-    blas'
-    lapack'
-    libsndfile
-    fftw
-    fftwSinglePrec
-    portaudio
-    qrupdate'
-    arpack'
-    libwebp
-    gl2ps
-  ]
-  ++ stdenv.lib.optionals enableQt [
-    qtbase
-    qtsvg
-    qscintilla
-  ]
-  ++ stdenv.lib.optionals (ghostscript != null) [ ghostscript ]
-  ++ stdenv.lib.optionals (hdf5 != null) [ hdf5 ]
-  ++ stdenv.lib.optionals (glpk != null) [ glpk ]
-  ++ stdenv.lib.optionals (suitesparse != null) [ suitesparse' ]
-  ++ stdenv.lib.optionals (enableJava) [ jdk ]
-  ++ stdenv.lib.optionals (sundials != null) [ sundials ]
-  ++ stdenv.lib.optionals (gnuplot != null) [ gnuplot ]
-  ++ stdenv.lib.optionals (python != null) [ python ]
-  ++ stdenv.lib.optionals (!stdenv.isDarwin) [ libGL libGLU libX11 ]
-  ++ stdenv.lib.optionals stdenv.isDarwin [
-    libiconv
-    darwin.apple_sdk.frameworks.Accelerate
-    darwin.apple_sdk.frameworks.Cocoa
-  ]
-  ;
-  nativeBuildInputs = [
-    pkgconfig
-    gfortran
-    # Listed here as well because it's outputs are split
-    fftw
-    fftwSinglePrec
-    texinfo
-  ]
-  ++ stdenv.lib.optionals (sundials != null) [ sundials ]
-  ++ stdenv.lib.optionals enableJIT [ llvm ]
-  ++ stdenv.lib.optionals enableQt [
-    qtscript
-    qttools
-  ]
-  ;
+  wrapOctave = callPackage ./wrap-octave.nix {
+    octave = self;
+    inherit (pkgs) makeSetupHook makeWrapper;
+  };
+
+  self = mkDerivation rec {
+    version = "6.1.0";
+    pname = "octave";
+
+    src = fetchurl {
+      url = "mirror://gnu/octave/${pname}-${version}.tar.gz";
+      sha256 = "0mqa1g3fq0q45mqc0didr8vl6bk7jzj6gjsf1522qqjq2r04xwvg";
+    };
+
+    buildInputs = [
+      readline
+      ncurses
+      perl
+      flex
+      qhull
+      graphicsmagick
+      pcre
+      fltk
+      zlib
+      curl
+      blas'
+      lapack'
+      libsndfile
+      fftw
+      fftwSinglePrec
+      portaudio
+      qrupdate'
+      arpack'
+      libwebp
+      gl2ps
+    ]
+    ++ stdenv.lib.optionals enableQt [
+      qtbase
+      qtsvg
+      qscintilla
+    ]
+    ++ stdenv.lib.optionals (ghostscript != null) [ ghostscript ]
+    ++ stdenv.lib.optionals (hdf5 != null) [ hdf5 ]
+    ++ stdenv.lib.optionals (glpk != null) [ glpk ]
+    ++ stdenv.lib.optionals (suitesparse != null) [ suitesparse' ]
+    ++ stdenv.lib.optionals (enableJava) [ jdk ]
+    ++ stdenv.lib.optionals (sundials != null) [ sundials ]
+    ++ stdenv.lib.optionals (gnuplot != null) [ gnuplot ]
+    ++ stdenv.lib.optionals (python != null) [ python ]
+    ++ stdenv.lib.optionals (!stdenv.isDarwin) [ libGL libGLU libX11 ]
+    ++ stdenv.lib.optionals stdenv.isDarwin [
+      libiconv
+      darwin.apple_sdk.frameworks.Accelerate
+      darwin.apple_sdk.frameworks.Cocoa
```

```suggestion
      Accelerate
      Cocoa
```

Please do inherit in the top-level with darwin.apple_sdk.frameworks.

---

Ping @doronbehar and @7c6f434c.